### DISCUSSION TITLE
Support other frameworks like Vue

### DISCUSSION THREAD

**Discussion by Aragur** [heroui-inc/heroui#790] [Ideas]
Foremost, many congratulations on this incredible UI library, it's extremely beautiful ❤️.
Currently, I'm locked in for using Vue in my projects and if I'm honest, I don't want to use React :)
Now I was thinking about porting these components to Vue, would this be permitted, or could we achieve a multi-framework UI system together?

**Comment by 50l3r** [comment]
I join the request. The library is very nice and feels great when interacting with the components. You can tell there is a lot of work behind it.

However, like @Aragur  all my efforts are focused on Vue and it is impossible for me to switch frameworks in my current projects.

I look forward to this becoming a reality 😁

**Comment by olehmelnyk** [comment]
hey, have you heard about mitosis? https://github.com/BuilderIO/mitosis
It looks like it can convert JSX-like components to other frameworks, including React Native, Angular, Vue, etc. - so MAYBE with it - it would be possible to make NextUI framework agnostic

**Reply by yxzlwz** [reply]
Emmmm, it seem not working with vue...

**Comment by ClearLuvMoki** [comment]
我觉得nextui是一个很好的库，他能做到在react组件库总脱颖而出，区别于antd/mui等风格，如果想在react中做到最好不应该去兼容其他框架（vue）

**Comment by yxzlwz** [comment]
Having the same proposal! I'm struggling to decide weather learn React for my new projects or give up this asesome UI

**Comment by I-am-abdulazeez** [comment]
Same here... Wanna work on a Vue project and I don't want to leave this NextxUI. It is beautiful.

**Comment by agileago** [comment]
same. it is the best ui library

**Comment by arshx86** [comment]
+++++ there is no library like nextui in vue (primevue maybe) it would be so great if someone try to port it!

**Comment by JenuelDev** [comment]
same here. Would love to use heroui in Vue.

**Comment by 50l3r** [comment]
Maybe Shadcn Vue? https://www.shadcn-vue.com/docs/components/alert.html

**Comment by yxzlwz** [comment]
Wow that's really a fantastic vue library! However I am planning to learn react now

**Comment by 50l3r** [comment]
I'm the same

Now I have to decide whether to use nextui or shadcn for react 😊

**Reply by yxzlwz** [reply]
Same here! When it comes to react, there's a lot of wonderful alternatives!

**Comment by I-am-abdulazeez** [comment]
Hello everyone! 👋 @Aragur, @olehmelnyk, @50l3r, @agileago, @JenuelDev 

We've created a repository and started working on the Vue port of HeroUI! 

Progress so far:

✅ Monorepo setup with pnpm + Turborepo
✅ Theme system (colors, tokens, design system)
✅ Core utilities and type system
✅ Button component (with variants, sizes, ripple effects)
✅ Spinner component
✅ Storybook/Histoire stories

[VeroUI](https://github.com/I-am-abdulazeez/veroui.)

We're following HeroUI's design system and bringing the same elegant experience to Vue 3. The project is still in early stages, but we'd love your support, feedback, and contributions! Feel free to:

⭐ Star the repo
🐛 Report issues
💡 Suggest features
🤝 Contribute components

Looking forward to building this together with the Vue community! 🚀

**Reply by wingkwong** [reply]
Please note that this project is not created by our HeroUI Team and the components will stick to v2. In fact, as we're moving to v3 and are making it framework agnostic. Hence, I don't think it's a good idea to start a new project at this time and I'd suggest to contribute to our repository instead so that our team could support it as well.

**Reply by I-am-abdulazeez** [reply]
Thanks for the feedback @wingkwong, I never said an "official port"... so, it is clear that it is a community build.

Thanks for the feedback, we'll make a PR, and create a vue folder under v3

**Reply by zkuljisatta** [reply]
404

**Reply by I-am-abdulazeez** [reply]
@zkuljisatta 

We did not reach agreement with the author. So, we've decided to build our own separately

https://vifui-test.vercel.app/

**Reply by hexadecimal233** [reply]
the repo is still private?

**Comment by xeno** [comment]
Highly recommend looking into [Nuxt UI](https://ui.nuxt.com). Nuxt is optional, and it's a beautiful Vue UI lib.
